packages/purgecss/src/index.ts

```diff
@@ -482,7 +485,13 @@ class PurgeCSS {
 
     // remove empty rules
     const parent = node.parent;
-    if (!node.selector) node.remove();
+    if (!node.selector) {
+      node.remove()
+      if (this.options.rejectedCss) {
+        node.selector = originalSelector;
+        this.removedNodes.push(node);
+      }
+    }
     if (isRuleEmpty(parent)) parent?.remove();
```

Comment:
There is the situation when you have an empty parent at the result of the node being removed. In that situation, I think you might want to add the parent node to the `removedNodes` array.

For:
```
@media (max-width: 66666px){
  .unused-class, .unused-class2 {
    color: black;
  }
}
```
We probably want to ouput the same for the rejectedCss, but it currently outputs:
```
rejectedCss: '\n  .unused-class, .unused-class2 {\n    color: black;\n  }'
```

Reply:
I'll add that test case and see how I can resolve it.

Reply:
@Ffloriel Thank you for your comment, it helped me understand a little more about how the code works. I added the following test case:
```css
@media (max-width: 66666px) {
  .used-class, .unused-class {
    color: black;
  }
}
```
Which I was handling incorrectly.
What the rejected css 'algorithm' now does is:
- keep track of what selectors are removed from the rule
- if any selector is removed from the rule, clone the rule (or its parent if there exists one)
- replace the (deemed critical) selectors from the clone with the rejected selectors
- preserve that rule node as `rejectedCss` output

So the above case would result in:
```css
/* critical */
@media (max-width: 66666px) {
  .used-class {
    color: black;
  }
}
/* rejected */
@media (max-width: 66666px) {
  .unused-class {
    color: black;
  }
}
```

Purging the css is easier since you walk down the AST nodes and remove what you don't need. Keeping the rejected css is harder since once you have a node you need to preserve you then also have to preserve its ancestors to get the correct output.

Can you think of any examples where a rule node is nested more that just 1 parent deep? My understanding is that these are CSS nodes thus the selectors are flattened, but if there is a case like:
```css
@media (min-width: 1px) {
  @media (max-width: 200px) {
    .selector {
      color: black;
    }
  }
}
```
I would have to add something that goes up the tree to preserve the correcct output

Reply:
Added a new PR that fixes the above [here](https://github.com/FullHuman/purgecss/pull/799)